lisa/runners/legacy_runner.py

```diff
@@ -459,11 +455,27 @@ def discover_running_cases(self) -> List[Dict[str, str]]:
             case_match = self.CASE_RUNNING.match(line)
             if case_match:
                 name = case_match["name"]
-                current_case: Dict[str, str] = case_match.groupdict()
+                current_case: Dict[str, str] = {
+                    key: value
+                    for key, value in case_match.groupdict().items()
+                    if value
+                }
             image_match = self.CASE_IMAGE_LOCATION.match(line)
             if image_match:
                 location = image_match["location"]
-                current_case.update(image_match.groupdict())
+                current_case.update(
+                    {
+                        key: value
+                        for key, value in image_match.groupdict().items()
+                        if value
+                    }
+                )
+                # gallery_image for ARMImage, vhd_image for OsVHD in legacy run
+                if "gallery_image" in current_case.keys():
+                    current_case["image"] = current_case.pop("gallery_image")
+                if "vhd_image" in current_case.keys():
```

Comment:
if one of them must be present, there can have an exception, if none of them found. 
if only one of them should be present, it can be else, or have a validation.